src/mam4xx/utils.hpp

```diff
@@ -209,7 +217,9 @@ void extract_stateq_from_prognostics(const mam4::Prognostics &progs,
   for (int m = 0; m < AeroConfig::num_modes(); ++m) {
     // First add the aerosol species mmr
     for (int a = 0; a < mam4::num_species_mode(m); ++a) {
-      q[s_idx] = progs.q_aero_i[m][a](klev);
+      const int prg_idx =
```

Comment:
I reviewed this part of the code and found that we cannot use ` mam4::mode_aero_species(mode, species_index)` to obtain the index of the species in each mode.  ` mam4::mode_aero_species(mode, species_index)` returns the aero_id, which is an enum from [here](https://github.com/eagles-project/mam4xx/blob/7d14b8a71166c48327f1839c46c0f974134a631f/src/mam4xx/aero_modes.hpp#L130).

Because mam4xx prognostics structure has same ordering of species per node that state_q and qqcw, we do not need to add this function; however, we could add use the function that @jeff-cohere have implemented [here](https://github.com/eagles-project/mam4xx/blob/7d14b8a71166c48327f1839c46c0f974134a631f/src/mam4xx/aero_modes.hpp#L249) and it is used [here](https://github.com/eagles-project/mam4xx/blob/7d14b8a71166c48327f1839c46c0f974134a631f/src/mam4xx/utils.hpp#L163).